meson.build

```diff
@@ -0,0 +1,301 @@
+'jets/f/cell.c',
+'jets/f/comb.c',
+'jets/f/cons.c',
+'jets/f/core.c',
+'jets/f/face.c',
+'jets/f/fitz.c',
+'jets/f/flan.c',
+'jets/f/flip.c',
+'jets/f/flor.c',
+'jets/f/fork.c',
+'jets/f/help.c',
+'jets/f/hike.c',
+'jets/f/look.c',
+'jets/f/loot.c'
+]
+
+jets_f_ut_src =  [
+'jets/f/ut.c',
+'jets/f/ut_burn.c',
+'jets/f/ut_buss.c',
+'jets/f/ut_conk.c',
+'jets/f/ut_crop.c',
+'jets/f/ut_find.c',
+'jets/f/ut_fire.c',
+'jets/f/ut_fish.c',
+'jets/f/ut_fuse.c',
+'jets/f/ut_gain.c',
+'jets/f/ut_lose.c',
+'jets/f/ut_mint.c',
+'jets/f/ut_mull.c',
+'jets/f/ut_nest.c',
+'jets/f/ut_peek.c',
+'jets/f/ut_peel.c',
+'jets/f/ut_play.c',
+'jets/f/ut_repo.c',
+'jets/f/ut_rest.c',
+'jets/f/ut_tack.c',
+'jets/f/ut_toss.c',
+'jets/f/ut_wrap.c'
+]
+
+jets_g_src = [
+'jets/g/down.c'
+]
+
+jets_src = [
+'jets/tree.c'
+]
+noun_src = ['noun/allocate.c',
+  'noun/events.c',
+  'noun/hashtable.c',
+  'noun/imprison.c',
+  'noun/jets.c',
+  'noun/manage.c',
+  'noun/nock.c',
+  'noun/retrieve.c',
+  'noun/trace.c',
+  'noun/vortex.c',
+  'noun/xtract.c',
+  'noun/zave.c']
+
+vere_src = ['vere/ames.c',
+  'vere/behn.c',
+  'vere/cttp.c',
+  'vere/http.c',
+  'vere/loop.c',
+  'vere/main.c',
+  'vere/raft.c',
+  'vere/reck.c',
+  'vere/save.c',
+  'vere/sist.c',
+  'vere/term.c',
+  'vere/time.c',
+  'vere/unix.c',
+  'vere/walk.c']
+
+src_list = [
+vere_src, noun_src,
+jets_a_src, jets_b_src,
+jets_c_src, jets_d_src,
+jets_e_src, jets_e_ed_src, jets_f_src, jets_f_ut_src,
+jets_g_src, jets_src]
+
+sources = []
+foreach s : src_list
+  sources += s
+endforeach
+
+incdir = include_directories('include/')
+
+conf_data = configuration_data()
+conf_data.set('URBIT_VERSION', '"0.5.1"')
+
+osdet = build_machine.system()
+os_c_flags = []
+os_deps = []
+os_link_flags = []
+
+if osdet == 'linux'
+  conf_data.set('U3_OS_linux', true)
+
+  pthread_dep = meson.get_compiler('c').find_library('pthread')
+  ncurses_dep = dependency('ncurses')
+  os_deps = os_deps + [pthread_dep, ncurses_dep]
+
+elif osdet == 'darwin'
+  conf_data.set('U3_OS_osx', true)
+
+  os_c_flags = os_c_flags + ['-bind_at_load']
+  #  os_link_flags = ['-framework CoreServices', '-framework CoreFoundation']
+  ncurses_dep = meson.get_compiler('c').find_library('ncurses')
+  os_deps = os_deps + [ncurses_dep]
+
+elif osdet == 'bsd'
+  conf_data.set('U3_OS_bsd', true)
+
+  pthread_dep = meson.get_compiler('c').find_library('pthread')
+  kvm_dep = meson.get_compiler('c').find_library('kvm')
+  ncurses_dep = dependency('ncurses')
+  os_deps = os_deps + [kvm_dep, pthread_dep, ncurses_dep]
+else
+  error('Unsupported OS detected:' + osdet)
+endif
+
+endian = build_machine.endian()
+
+if endian == 'little'
+  conf_data.set('U3_OS_ENDIAN_little', true)
+else
+  error('Little endian encoding required')
+endif
+
+configure_file(input : 'include/config.h.in',
+               output : 'config.h',
+               configuration : conf_data)
+
+# We expect these libs to supplied with the distribution
+openssl_dep = dependency('openssl')
```

Comment:
Do we not need to pin the versions of openssl, libuv, and curl? It feels a little dangerous to me to let them float based on the platform.